### PR TITLE
Block user's account.

### PR DESCRIPTION
**Pull request by teoperisanu**
Check the number of continuous failed login attempts. In case of more than 3 attempts, block the user account.

Signed-off-by: teoperisanu <teodosie.perisanu@gmail.com>

### CODE REVIEW

File: teamup-server/src/main/java/com/project/teamup/security/JwtUserDetailsService.java [hunk 1]
```diff
@@ -15,10 +16,15 @@
 public class JwtUserDetailsService implements UserDetailsService {
     @Autowired
     private UserRepository userRepository;
+    @Autowired
+    private LoginAttemptService loginAttemptService;
 
     @Override
     public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
         Optional<com.project.teamup.model.User> user = userRepository.findByUsername(username);
+        if (loginAttemptService.isBlocked(username)) {
```
Comment (popiosifcristian) [inline teamup-server/src/main/java/com/project/teamup/security/JwtUserDetailsService.java]: Please change the username parameter with the user found above and move your logic in the if statement declared below

File: teamup-server/src/main/java/com/project/teamup/security/LoginAttemptService.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            }
+            user.setFailedLoginAttempts(user.getFailedLoginAttempts() + 1);
+            userRepository.save(user);
+        } else {
+            throw new UsernameNotFoundException("User not found");
+        }
+    }
+
+    public boolean isBlocked(String username) {
+        Optional<com.project.teamup.model.User> tempUser = userRepository.findByUsername(username);
```
Comment (popiosifcristian) [inline teamup-server/src/main/java/com/project/teamup/security/LoginAttemptService.java]: As mentioned above

File: teamup-server/src/main/java/com/project/teamup/security/JwtUserDetailsService.java (hunk 1 above)
Comment (teoperisanu) [reply teamup-server/src/main/java/com/project/teamup/security/JwtUserDetailsService.java]: Fixed in commit e196aa0.

File: teamup-server/src/main/java/com/project/teamup/security/LoginAttemptService.java (hunk 2 above)
Comment (teoperisanu) [reply teamup-server/src/main/java/com/project/teamup/security/LoginAttemptService.java]: Modified in commit e196aa0.

**Pull request merged**